modules/shortlinks.php

```diff
@@ -97,12 +97,26 @@ function wpme_get_shortlink_handler( $shortlink, $id, $context, $allow_slugs ) {
  * @uses register_rest_field, wpme_rest_get_shortlink
  */
 function wpme_rest_register_shortlinks() {
+	
+	/**
+	 * Filter the post type list for which shortlinks are allowed.
+	 *
+	 * @module shortlinks
+	 *
+	 * @since 7.7.0
+	 *
+	 * @param array Array of post types.
+	 */	
+	$allowed_post_types = apply_filters( 'jetpack_shortlink_allowed_post_types',
```

Comment:
I think I would change the name of that filter as well as its description. When reading `jetpack_shortlink_allowed_post_types `, one can assume that adding a post type to that filter is enough for that post type to work well with Shortlinks. 

It's not enough though; the post type also has to declare support for the shortlinks, as shown here:
https://github.com/Automattic/jetpack/blob/9654910469179fc8c92a0e49d35cedeb153ef5a0/modules/shortlinks.php#L75
https://jetpack.com/support/wp-me-shortlinks/#cpt

The filter we're adding here will really only be useful to display the shortlinks interface in the block editor for a specific post type:
![image](https://user-images.githubusercontent.com/426388/62627461-bf8d0a80-b929-11e9-81c4-00542ae5e377.png)

I think the filter name and description should reflect that.

Reply:
Since it already depends on `post_type_support` could we change the array being registered with `array_merge( array( 'post' ), get_post_types_by_support( 'shortlinks' ) );` ?